packages/gafl-webapp-service/src/processors/__tests__/payment.spec.js

```diff
@@ -0,0 +1,115 @@
+import { preparePayment } from '../payment.js'
+import { licenceTypeAndLengthDisplay } from '../licence-type-display.js'
+
+jest.mock('../licence-type-display.js');
+licenceTypeAndLengthDisplay.mockReturnValue('Trout and coarse, up to 2 rods, 8 day')
+
+const createRequest = () => ({
+  headers: {
+    'x-forwarded-proto': 'https'
+  },
+  info: { host: 'localhost:1234' },
+  server: { info: { protocol: 'http' } },
+})
+
+const createTransaction = () => ({
+  id: 'transaction-id',
+  cost: 12,
+  permissions: [{ 
+    licensee: {
+      firstName: 'Lando',
+      lastName: 'Norris',
+      email: 'test@example.com',
+      premises: '4',
+      street: 'Buttercup lane',
+      locality: 'Clifton',
+      postcode: 'BS8 3TP',
+      town: 'Bristol',
+      country: 'GB-ENG'
+    }
+  }]
+})
+
+describe('preparePayment', () => {
+  let request, transaction, result
+  beforeEach(() => {
+    request = createRequest()
+    transaction = createTransaction()
+    result = preparePayment(request, transaction)
+  })
+  describe('provides the correct return url', () => {
+    it('when the "x-forwarded-proto" header is present', () => {
+      expect(result.return_url).toBe('https://localhost:1234/buy/agreed')
```

Comment:
After looking for a bit, I can see that `result.return_url` should be the same as ` ``http://${request.info.host}/buy/agreed`` `, however a bit more of a thorough test would be a bit more descriptive. If you were to modify the `createRequest` method to allow for host to be specified, but default to `localhost:1234`, you could then do something like:
```
['localhost:4321', 'otherhost:8888', 'samplehost:4444'].forEach(host => 
  it(`uses SSL when 'x-forwarded-proto' header is present, host '${host}'`, () => {
    const request = createRequest(host)
    expect(result.return_url).toBe(`https://${host}/buy/agreed`)
  })
)
```